libs/langchain-community/src/llms/watsonx_ai.ts

```diff
@@ -127,74 +83,50 @@ export class WatsonxAI extends LLM<BaseLLMCallOptions> {
     prompt: string,
     _options: this["ParsedCallOptions"]
   ): Promise<string> {
-    interface WatsonxAIResponse {
-      results: {
-        generated_text: string;
-        generated_token_count: number;
-        input_token_count: number;
-      }[];
-      errors: {
-        code: string;
-        message: string;
-      }[];
-    }
-    const response = (await this.caller.call(async () =>
-      fetch(this.endpoint, {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-          Accept: "application/json",
-          Authorization: `Bearer ${await this.generateToken()}`,
-        },
-        body: JSON.stringify({
-          project_id: this.projectId,
-          model_id: this.modelId,
-          input: prompt,
-          parameters: this.modelParameters,
-        }),
-      }).then((res) => res.json())
-    )) as WatsonxAIResponse;
-
-    /**
-     * Handle Errors for invalid requests.
-     */
-    if (response.errors) {
-      throw new Error(response.errors[0].message);
-    }
-
-    return response.results[0].generated_text;
+    return await this.caller.call(async () =>
+      this.watsonApiClient.generateText(
+        prompt,
+        this.projectId,
+        this.modelId,
+        this.modelParameters
+      )
+    );
   }
 
-  async generateToken(): Promise<string> {
-    if (this.ibmCloudToken && this.ibmCloudTokenExpiresAt) {
-      if (this.ibmCloudTokenExpiresAt > Date.now()) {
-        return this.ibmCloudToken;
-      }
-    }
-
-    interface TokenResponse {
-      access_token: string;
-      expiration: number;
-    }
-
-    const urlTokenParams = new URLSearchParams();
-    urlTokenParams.append(
-      "grant_type",
-      "urn:ibm:params:oauth:grant-type:apikey"
+  async *_streamResponseChunks(
```

Comment:
I may be missing something, but it seems like while we have the stream function for the LLM it isn't actually usable in the current state?

I was expecting something closer to the [sagemaker implementation](https://github.com/langchain-ai/langchainjs/blob/88713ea81f3d015e0c813c19cd63a9be0bf5cd68/libs/langchain-community/src/llms/sagemaker_endpoint.ts#L180-L188) which would allow easy toggling of streaming.

Though the present `_call` function appears to directly call `generate_text`.

Reply:
I was originally inspired by the way the ollama connector handles it, but let me take another look at this. I only tested the chat model I implemented for streaming and invoking.